internal/boxcli/setup.go

```diff
@@ -39,3 +42,35 @@ func runInstallNixCmd(cmd *cobra.Command) error {
 	}
 	return nix.Install()
 }
+
+func ensureNixInstalled(cmd *cobra.Command, args []string) error {
+	if nix.BinaryInstalled() {
+		return nil
+	}
+	if nix.DirExists() {
+		if err := nix.SourceNixEnv(); err != nil || nix.BinaryInstalled() {
+			return err
+		}
+
+		return usererr.New(
+			"We found a /nix directory but nix binary is not in your PATH and we " +
+				"were not able to find it in the usual locations. Your nix installation " +
+				"might be broken. If restarting your terminal or reinstalling nix " +
+				"doesn't work, please create an issue at " +
+				"https://github.com/jetpack-io/devbox/issues",
+		)
+	}
+
+	color.Yellow("\nNix is not installed. Devbox will attempt to install it.\n\n")
+
+	if skipPrompts, _ := cmd.Flags().GetBool(globalYesFlag); !skipPrompts {
```

Comment:
Can we simply skip this if the terminal is non-interactive?

Reply:
will do!